[#22] 결제 관리

woodk3268 opened the pull request:
### 4. 결제 시스템 (Payment System)

#### 4.1 결제 수단

- **일반 결제**: 신용카드, 체크카드, 계좌이체, 무통장입금
- **간편 결제**: 토스페이먼츠, 카카오페이
- **포인트**: 야놀자 적립금

#### 4.2 쿠폰/할인

- **쿠폰 유형**: 정액 할인, 정률 할인
- **적용 조건**: 최소 주문 금액, 최대 할인 금액, 유효기간
- **발급 방식**: 회원가입, 이벤트 참여, 리뷰 작성, 생일 쿠폰, 수동 발급

#### 4.3 적립금/포인트

- **적립 정책**: 결제 금액의 1% (기본 적립률)
- **사용 규칙**: 주문 시 포인트 차감, 주문 취소 시 자동 복원
- **유효기간**: 적립일로부터 2년

#### 4.4 결제 처리

- **결제 상태**: 대기, 성공, 실패, 취소, 환불, 부분환불
- **환불 지원**: 전액 환불, 부분 환불

woodk3268 committed:
add: 결제 엔티티 추가

woodk3268 committed:
add: 결제 방식 enum 추가

woodk3268 committed:
add: 결제 상태 enum 추가

woodk3268 committed:
modify : db schema 변경

1. 주문, 포인트, 주문 쿠폰 테이블 추가
2. 결제, 쿠폰, 사용자 쿠폰 테이블 schema 변경

woodk3268 committed:
add : 주문 엔티티 추가

woodk3268 committed:
add : 주문 쿠폰 엔티티 추가

woodk3268 committed:
add : 주문 상태 enum 추가

woodk3268 committed:
add : 포인트 엔티티 추가

woodk3268 committed:
add : 포인트 상태 enum 추가

woodk3268 committed:
add : 포인트 거래 유형 enum 추가

woodk3268 committed:
add : 쿠폰 엔티티 추가

woodk3268 committed:
add : 쿠폰 발급 유형 enum 추가

woodk3268 committed:
add : 쿠폰 상태 enum추가

woodk3268 committed:
add : 사용자 쿠폰 엔티티 추가

woodk3268 committed:
add : 사용자 쿠폰 상태 enum 추가

woodk3268 committed:
add : 할인 유형 enum 추가

woodk3268 committed:
add : 주문 엔티티에 주문 관련 메서드 추가

woodk3268 committed:
add : 주문 쿠폰 엔티티에 쿠폰ID 조회 메서드 추가

woodk3268 committed:
add : 쿠폰 엔티티에 사용된 수량 증가 메서드 추가

woodk3268 committed:
add : 주문 dto 추가

woodk3268 committed:
add : 주문 쿠폰 dto 추가

woodk3268 committed:
modify : 주문 쿠폰 dto에 사용자 쿠폰 id 필드 추가

woodk3268 committed:
add : 주문 repository 추가

woodk3268 committed:
add : 주문 쿠폰 repository 추가

woodk3268 committed:
add : 쿠폰 repository 추가

woodk3268 committed:
add : 사용자 쿠폰 repository 추가

woodk3268 committed:
add : 포인트 repository 추가

woodk3268 committed:
add : 결제 repository 추가

woodk3268 committed:
add : 결제 dto 추가

woodk3268 committed:
결제 관련 entity, repository 패키지 위치 변경

woodk3268 committed:
결제 관련 entity, repository 패키지 위치 변경

woodk3268 committed:
add : 쿠폰 controller 추가

woodk3268 committed:
add : 포털용 쿠폰 controller 추가

woodk3268 committed:
delete : 쿠폰 repository 안 쓰는 쿼리 삭제

woodk3268 committed:
add : 쿠폰, 사용자-쿠폰 엔티에 동시성 고려 코드 추가

woodk3268 committed:
add : 쿠폰 dto 추가

woodk3268 committed:
add : Redis 분산 락 서비스 추가

woodk3268 committed:
add : 쿠폰 service, test 코드 추가

woodk3268 committed:
add : 포인트 dto 추가

woodk3268 committed:
add : 포인트 controller 추가

woodk3268 committed:
add : 포털용 포인트 controller 추가

woodk3268 committed:
add : 포인트 repository 불필요한 코드 삭제

woodk3268 committed:
add : 포인트 service, test 추가

woodk3268 committed:
add : 주문 entity에 동시성 관련 코드 추가

woodk3268 committed:
add : 주문 controller 추가

woodk3268 committed:
delete : 주문 repository 불필요한 코드 삭제

woodk3268 committed:
delete : 주문 service, test 추가

woodk3268 committed:
add : application.yml 에 pg 결제 설정 추가

woodk3268 committed:
add : pg 결제 설정 config 파일 추가

woodk3268 committed:
add : PG사 API 호출 위해 webconfig 추가

woodk3268 committed:
add : 결제 엔티티에 동시성 관련 코드 추가

woodk3268 committed:
add : 결제, 결제 통계 dto 추가

woodk3268 committed:
add : 결제 controller 추가

woodk3268 committed:
delete : 결제 repository 불필요한 코드 삭제

woodk3268 committed:
add : pgService 추가

woodk3268 committed:
add : 결제 service, test 추가

woodk3268 committed:
add : 결제 SchedulerService 추가

woodk3268 committed:
add : 리뷰, 리뷰 이미지 entity 추가

woodk3268 committed:
add : 리뷰 dto 추가

woodk3268 committed:
delete : ApiResponse 성공 응답시, 메시지 포함한 메서드 제거

woodk3268 committed:
add : 리뷰 dto 추가

woodk3268 committed:
add : 리뷰 entity 내용 추가

woodk3268 committed:
add : 리뷰 reposiotry 추가

woodk3268 committed:
add : 리뷰 이미지 repository 추가

woodk3268 committed:
add : 리뷰 컨트롤러 추가

woodk3268 committed:
add : 리뷰 service, test 추가

woodk3268 committed:
add : UserDetail 에 getId() 메서드 추가

woodk3268 committed:
modify : 컨트롤러 성공 응답시, 메시지 포함해서 반환하는 코드 수정

woodk3268 committed:
add : 리뷰 controller에 권한 확인 코드 추가

woodk3268 committed:
modify : 숙소 엔티티 평점 업데이트, 리뷰 수 감소 메서드 추가 및 리뷰 수 증가 메서드 변경

woodk3268 committed:
add: 리뷰 작성 시, 숙소의 리뷰 수와 평점 업데이트

woodk3268 committed:
add: 리뷰 작성 시, 숙소의 리뷰 수와 평점 업데이트 test

mason1155 reviewed (commented):
결제는 잘 되던가요??
프론트 페이지가 필요하지 않나요?

woodk3268 committed:
Redis 분산 락을 이용한 동시성 제어 구현

- common 패키지에 통합 DistributedLockService 구현
- 리뷰 생성/수정/삭제 시 분산 락 적용으로 평점 계산 동시성 보장
- Payment 패키지의 기존 분산 락 서비스 통합 및 중복 제거
- Redis 설정 추가 및 락 타임아웃/리스 시간 설정

woodk3268 committed:
JPA 엔티티 매핑 오류 수정 (OrderCoupon, PaymentRepository)

woodk3268 committed:
테스트 코드 DistributedLockService import문 추가

woodk3268 committed:
테스트 코드 - JPA 엔티티 매핑 오류 수정

woodk3268 commented:
현재 프론트엔드 결제 페이지 구현 중이고, 완성되는 대로 PAYCO 테스트 환경에서 결제 프로세스 전체를 테스트해보겠습니다.

woodk3268 committed:
Merge pull request #25 from f-lab-edu/feature/24

[#24] 리뷰 관리

woodk3268 committed:
Merge branch 'develop' into feature/22
